auth/README.md

```diff
@@ -0,0 +1,133 @@
+# Authorization Interceptor
+
+This package provides a server-side gRPC interceptor that interfaces with Themis, a policy engine that is developed and maintained by Infoblox. It is designed to give developers fine-grained control over who (e.g. specific users) or what (e.g. neighboring services) can access their gRPC service's business logic.
+
+## How it Works
+When building a gRPC server, a developer can supply his or her server with a collection of _gRPC server-side interceptors_. Each request that's sent to the gRPC server will traverse the gRPC interceptors before reaching the server's business logic. At any point, an interceptor can stop a given request from advancing to the application.
+
+Here are some common usages for gRPC interceptors.
+
+- Logging (e.g. log to `stdout` on each request)
+- Metrics (e.g. increment a request-counting metric)
+- **Authorization** (e.g. check if the request-sender is allowed to access a given resource)
+
+When the authorization interceptor is enabled, it ensures that whoever or whatever sent the request is allowed to access some endpoint. If they're not, the interceptor issues a `DENY` response, otherwise the request moves to the application itself.
+
+Behold the behavior authorization interceptor as if it were a living, breathing human: _"Hey Themis, is the sender of request XYZ allowed to access endpoint ABC in my service? They're not? Okay, I'll deny the request!"_
+
+## Add Interceptors
+If you already use gRPC with Go, your project probably has a bit of code like this.
+
+```go
+// Define your grpc server
+myServer := grpc.NewServer()
+```
+This is how you would decorate your server with interceptors.
+
+```go
+// Define your grpc server
+myServer := grpc.NewServer(
+  rpc.StreamInterceptor(grpc_middleware.ChainStreamServer(
+    // Define your list of grpc interceptors
+  )),
+)
+```
+
+For more information about adding server-side gRPC interceptors, check out [the documentation](https://github.com/grpc-ecosystem/go-grpc-middleware) on the official gRPC GitHub repository.
+
+## Add the Default AuthZ Interceptor
+
+The `auth` package offers a default authorization interceptor.
+```go
+themisAddress := "default.themis:5555" 
+applicationID := "shopping-mall"
+// Define your grpc server
+myServer := grpc.NewServer(
+  grpc.UnaryInterceptor(grpc_middleware.ChainStreamServer(
+    // Include the authorization interceptor
+    auth.UnaryServerInterceptor(themisAddress, applicationID),
+  )),
+)
+```
+In the example above, the `themisAddress` variable corresponds to the host and port of Themis. The `applicationID`  maps the interceptor to a specific application. In a microservices environment, you might have multiple services that, as a unit, compose an application (e.g. a "petstore" service, a "coffee shop" service, and a "cellphone kiosk" service might belong to a "shopping mall" application). Each service has its own access control logic, but they're all under the same application umbrella.
```

Comment:
is there any open-source info available on the web for this application id stuff? Is there any plan to?

Reply:
No, I don't believe so. From what I know, the notion of an application identifier and how/where it would get used isn't well known 😕

John wrote-up a Google Doc about it a while ago, but I don't think it's progressed much since then.

Reply:
this is kind of a puzzling feature, then, to have in an open source repository ;)